ngraph/core/src/op/pad.cpp

```diff
@@ -200,16 +200,35 @@ bool op::v1::Pad::evaluate_pad(const HostTensorVector& outputs, const HostTensor
     } else {
         pad_value = pad_zero_value.data();
     }
+
+    // compute pads_begin and pads_end CoordinateDiffs from pads_begin 
```

Comment:
@ilyachur I guess the logic of `validate_and_infer_types()` should be revised due it relies on `get_pads_begin()` (and _end) which work correctly only for Constant ops (so I propose to remove or extend `get_pads_begin()` (and _end))

Reply:
We cannot remove these method because it break the backward compatibility.
Moreover, `validate_and_infer_types()` can work only with constant operations, because we don't have data in the validation moment.

Reply:
After review with @GlebKazantaev, we decided to revert previous changes and set op input tensor to op output port (https://github.com/openvinotoolkit/openvino/pull/8697/files#diff-3935ba3e8c72b38d9878d407248d5bccc507f293d89c8f9d13fc29a28056b7d7R108). In this case, `get_pads_begin()` treats Parameter as Const and deduce shape correctly on `validate_and_infer_types()` before `evaluate_pad()`

Reply:
After offline discussion, decided to return to previous fix inside `evaluate_pad`